src/layer/tile/GridLayer.js

```diff
@@ -343,10 +343,15 @@ L.GridLayer = L.Layer.extend({
 	},
 
 	_setView: function (center, zoom, noPrune, noUpdate) {
-		var tileZoom = Math.round(zoom),
-			tileZoomChanged = this._tileZoom !== tileZoom;
+		var tileZoom = Math.round(zoom);
+		if ((this.options.maxZoom !== undefined && tileZoom > this.options.maxZoom) ||
+		    (this.options.minZoom !== undefined && tileZoom < this.options.minZoom)) {
+			tileZoom = undefined;
+		}
+
+		var tileZoomChanged = (tileZoom !== this._tileZoom);
 
-		if (!noUpdate && tileZoomChanged) {
+		if (!noUpdate || tileZoomChanged) {
```

Comment:
If animating, ignore the `noUpdate` flag and reload tiles anyway if we're switching the grid layer to another level. This way there's no need to throttle!